lib/Target/SubstraitPB/Export.cpp

```diff
@@ -45,20 +45,51 @@ DECLARE_EXPORT_FUNC(RelOpInterface, Rel)
 
 FailureOr<std::unique_ptr<proto::Type>> exportType(Location loc,
                                                    mlir::Type mlirType) {
-  // TODO(ingomueller): Support other types.
-  auto si32 = IntegerType::get(mlirType.getContext(), 32, IntegerType::Signed);
-  if (mlirType != si32)
-    return emitError(loc) << "could not export unsupported type " << mlirType;
+  MLIRContext *context = mlirType.getContext();
+
+  // Handle SI1.
+  auto si1 = IntegerType::get(context, 1, IntegerType::Signed);
+  if (mlirType == si1) {
+    // TODO(ingomueller): support other nullability modes.
+    auto i1Type = std::make_unique<proto::Type::Boolean>();
+    i1Type->set_nullability(
+        Type_Nullability::Type_Nullability_NULLABILITY_REQUIRED);
+
+    auto type = std::make_unique<proto::Type>();
+    type->set_allocated_bool_(i1Type.release());
+    return std::move(type);
```

Comment:
Is there a way we could avoid make_unique's here? (well more the memory allocation parts here if this done often).

Reply:
You are touching upon a question I had pending in my head: what is the best practice to assemble nested protobufs. I searched both internal and external doc and couldn't find any clear answers.

It sound *plausible* that using `std::make_unique` and `set_allocated_*` entails more allocations than necessary. I *suppose* that the better alternative is to use something like:

```c++
Inner* inner = outer->mutable_inner();
inner->set_primite(...);
```

If that is the case, then I'd have to restructure the whole file, so I'd like to be sure what is the best approach before starting. Can you confirm or have a good resource on the top of your head?

Reply:
Resolving conversation here in favor of #826.